app/helpers/steps_helper.rb

```diff
@@ -19,21 +25,45 @@ module StepsHelper
     }.freeze,
   }.freeze
 
+  VEHICLE_RULES = {
+    vehicle_not_owned: %i[vehicle].freeze,
+    vehicle_owned: FIRST_VEHICLE_STEPS,
+    vehicle_regular: ALL_VEHICLE_STEPS,
+  }.freeze
+
   def next_step_for(intro, step)
-    next_estimate_step(steps_list_for(intro), step)
+    next_estimate_step(steps_list_for(intro).flatten, step)
+  end
+
+  def last_step?(model, step)
```

Comment:
I don't like this method name, because it made me assume this was referring to the last step overall, not the last step in the current subgroup. `last_step_in_group?` ? `last_step_in_current_list?`